src/process/process.rs

```diff
@@ -97,19 +98,30 @@ pub struct Process<'a> {
 }
 
 #[inline(never)]
-pub unsafe fn load_processes(mut start_addr: *const usize) ->
+pub unsafe fn load_processes(start_addr: *const u8) ->
         &'static mut [Option<Process<'static>>] {
-    for op in PROCS.iter_mut() {
-        if *start_addr != 0 {
-            let prog_start = start_addr.offset(1);
-            let length = *start_addr as isize;
-            start_addr = (start_addr as *const u8).offset(length) as *const usize;
-
-            *op = Process::create(prog_start as *mut u8, length);
-        } else {
-            *op = None;
+    let mut addr = start_addr;
+    let mut process_iter = PROCS.iter_mut();
+
+    loop {
+        // The first member of the LoadInfo header contains the total size of each process image. A
+        // sentinel value of 0 (invalid because it's smaller than the header itself) is used to
+        // mark the end of the list of processes.
+        let total_size = *(addr as *const usize);
```

Comment:
Could actually already cast it to a `LoadInfo` here:

```rust
let load_info = &*(addr as *const LoadInfo);
if load_info.total_size == 0 {
    break;
}
```

Not a big difference for now, I think it might make this code slightly less magical. I suspect it'll become more useful to do this when this gets refactored, but just a weak suggestion for now.